ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -10672,8 +10672,15 @@ ABoundActivity * HqlCppTranslator::doBuildActivityOutputIndex(BuildCtx & ctx, IH
     if (expr->hasAttribute(maxLengthAtom))   flags.append("|TIWmaxlength");
     if (compressAttr)
     {
-        if (compressAttr->hasAttribute(rowAtom))   flags.append("|TIWrowcompress");
-        if (!compressAttr->hasAttribute(lzwAtom))  flags.append("|TIWnolzwcompress");
+        if (!compressAttr->queryChild(0)->isAttribute())
+        {
+            flags.append("|TIWcompressdefined|TIWnolzwcompress");
+            doBuildVarStringFunction(instance->startctx, "queryCompression", compressAttr->queryChild(0));
+        }
+        else if (compressAttr->hasAttribute(rowAtom))   flags.append("|TIWrowcompress");
+        else if (compressAttr->hasAttribute(firstAtom))  flags.append("|TIWnolzwcompress");
```

Comment:
This has changed the output.  Previously this flag was also output if rowAtom was set.  See ecl/regress/ghoogle4.ecl

Reply:
ghoogle4.ecl does not compile for me - is there some trick to it?

Reply:
I have pushed a change to address this